New version

ellabeck03 opened the pull request.

py1sl commented:
need to remove all the .pyc and vscode extra files.
need to update your fork so it accounts for the changes already made to neut constants.
looks like the issues with the test data were windows using the opposite slash compared to linux, so you proposed use of os.path objects is probably the correct solution.

currently there is a test fail as you need to replace the old tests with updated  ones using the new function.
after that i will do a bit more of a code review, its not clear that the changes to the cooling step part are quite right.

ellabeck03 commented:
Hi, I have replaced the old plotting tests and that now seems to be working fine. Would you like me to replace paths with os.path objects any time I spot a FileNotFound error caused by them? I did it for the test_fispact_analysis file just to confirm that every test in the file passed, and it all worked well. I am fairly sure I have updated the fork and removed all .pyc files created by running the tests (currently researching a way to stop these being generated). I determined whether or not a sample was cooling based on the "Irradn/Cooling" labels generated in the first column of the output summary table, but please let me know if there is a better way to do this and I will implement it :)

ellabeck03 commented on fispact_analysis.py:
```diff
@@ -162,18 +140,20 @@ def plot_dose(sum_dat, offset=0, fname=None,
     if vlines:
         for xc in vlines:
-            plt.vline(x=xc + offset)
+            plt.axvline(x=xc + offset)
```
![image](https://github.com/user-attachments/assets/50b7c804-d8f3-43e1-ae39-353f662b34a1)
vline was generating an error, i googled axvline and i'm pretty sure it does the same thing? i can change this back if preferred :)

py1sl replied:
my understanding is that axvline can only take values between 0 and 1, whilst vlines isany value,  perhaps it is just a typo and should be vlines

ellabeck03 replied:
Yes sorry, you're absolutely right - I'll change this to vlines now!

ellabeck03 commented on fispact_output_reader.py:
```diff
@@ -223,7 +223,7 @@ def read_summary_data(data):
         cool_str = " -----Irradiation Phase-----"
     else:
-        cool_str = "  COOLING STEPS"
+        cool_str = "-----Cooling Phase-----"
```
Thank you for letting me know, I didn't realise this and I thought it had been missed when I was examining the output file, will change this back now :)

ellabeck03 commented on fispact_output_reader.py:
```diff
@@ -241,10 +241,18 @@ def read_summary_data(data):
 
     for line in sum_lines:
         if isFisII(data):
+            if "Irradn" in line:
```
This should be finished now, thank you for clarifying. Pandas really didn't like me adding only selected time steps to the data frame (since all columns need to be the same length), so I've stored any timesteps from the irradiation phase as Nan in the cooling phase timesteps column. Hopefully this shouldn't affect the plotting in any way. I have also created a test to see that it was putting Nans and actual numbers in the right indices within the column and it all worked as expected :)

py1sl replied:
definately need to read all the time steps, the selection to only cooling should be done later. best practice for this sort of file reading is to read everything into appropriate data structures then apply filtering later as needed.

ellabeck03 replied:
![image](https://github.com/user-attachments/assets/41074237-c225-4245-8ed3-83b0b2767e32)
Sorry, I think I need a bit more clarification on how best to approach this. Should I instead create more data frames with only data from the cooling/irradiation phase in them respectively? That way you read and retain the original data frame, with the option to plot one phase at a time. Also, there is a data point within the irradiation phase that is classified as cooling (image attached), and I'm not sure how to classify this if I was to create separate data frames.
